code/SubsitesVirtualPage.php

```diff
@@ -230,4 +230,12 @@ public function init()
         
         Subsite::$disable_subsite_filter = $origDisableSubsiteFilter;
     }
+    
+    
+    public function getViewer($action)
+    {
+        $controller = ModelAsController::controller_for($this->CopyContentFrom());
+        $controller->init();
```

Comment:
I'd be weary of calling `init` - we don't do this on `VirtualPage` (https://github.com/silverstripe/silverstripe-cms/blob/3.5/code/model/VirtualPage.php#L498-L502)